modules/cactus-web/src/helpers/positionPopover.ts

```diff
@@ -17,3 +18,107 @@ export function getTopPosition(targetRect: PRect, popoverRect: PRect): { top: st
   }
 }
+
+type RefOrId = RefObject<HTMLElement | string | null>
+export type PositionCallback = (elem: HTMLElement, anchor: HTMLElement | null) => void
+
+const getElem = ({ current }: RefOrId): HTMLElement | null =>
+  typeof current === 'string' ? document.getElementById(current) : current
+
+interface UsePositioning {
+  position: PositionCallback
+  visible: boolean
+  ref: RefOrId
+  anchorRef?: RefOrId
+  updateOnScroll?: boolean
+}
+
+export function usePositioning({
+  position,
+  visible,
+  ref,
+  anchorRef,
+  updateOnScroll = false,
+}: UsePositioning): void {
+  useLayoutEffect(() => {
+    if (visible && position && ref.current) {
+      const elem = getElem(ref)
+      if (elem) {
+        const anchor = anchorRef ? getElem(anchorRef) : null
+        position(elem, anchor)
+        if (updateOnScroll) {
+          return positionOnScroll(position, ref, anchorRef)
+        }
+      }
+    }
+  }, [visible, position, updateOnScroll, ref, anchorRef])
+}
+
+function positionOnScroll(position: PositionCallback, ref: RefOrId, anchorRef?: RefOrId) {
+  let updating = false
+  const listener = (e: Event) => {
+    if (!updating) {
+      updating = true
+      const target = e.target as Node | null
+      window.requestAnimationFrame(() => {
+        const anchor = anchorRef ? getElem(anchorRef) : null
+        if (!anchor || !target?.contains || target.contains(anchor)) {
+          const elem = getElem(ref)
+          if (elem) {
+            position(elem, anchor)
+          }
+        }
+        updating = false
+      })
+    }
+  }
+  window.addEventListener('scroll', listener, { passive: true, capture: true })
+  return () => window.removeEventListener('scroll', listener, true)
+}
+
+export const positionDropDown: PositionCallback = (dd, button) => {
+  if (!button) return
+  const ddRect = dd.getBoundingClientRect()
+  const { left, right, top, bottom } = button.getBoundingClientRect()
+  const view = getViewport() as Element
+  const maxRight = view.clientWidth
+  const maxBottom = view.clientHeight
+
+  const minWidth = Math.min(maxRight, Math.round(right - left))
+  const maxWidth = Math.min(maxRight, minWidth + Math.min(minWidth, 400))
+  const ddWidth = Math.round(ddRect.width)
+  const ddHeight = Math.round(ddRect.height) + MARGIN
+  const expectedWidth = Math.max(minWidth, Math.min(maxWidth, ddWidth))
+
+  // Assumes fixed positioning.
+  const expectedRight = left + expectedWidth
+  if (expectedRight > maxRight) {
+    dd.style.left = ''
+    dd.style.right = `${Math.min(maxRight - right, 0)}px`
+  } else {
+    dd.style.right = ''
+    dd.style.left = `${left}px`
+  }
+
+  const expectedBottom = bottom + ddHeight
+  const expectedTop = top - ddHeight
+  if (expectedBottom > maxBottom && expectedTop > 0) {
+    dd.style.top = ''
+    dd.style.bottom = `${maxBottom + MARGIN - top}px`
+  } else {
+    dd.style.bottom = ''
+    dd.style.top = `${bottom + MARGIN}px`
+  }
```

Comment:
What if we deleted the `max-height` defined in `Select.tsx` and changed this section to:

```suggestion
  if (expectedBottom > maxBottom && expectedTop > 0) {
    const styleBottom = maxBottom + MARGIN - top
    dd.style.top = ''
    dd.style.bottom = `${styleBottom}px`
    dd.style.maxHeight = `${view.clientHeight - styleBottom - 10}px`
  } else {
    const styleTop = bottom + MARGIN
    dd.style.bottom = ''
    dd.style.top = `${styleTop}px`
    dd.style.maxHeight = `${view.clientHeight - styleTop - 10}px`
  }
```